tough/src/editor/signed.rs

```diff
@@ -494,3 +504,187 @@ impl SignedRepository {
         Ok(())
     }
 }
+
+impl TargetsWalker for SignedDelegatedTargets {
+    fn targets(&self) -> HashMap<String, &Target> {
+        let mut targets_map = HashMap::new();
+        for targets in &self.roles {
+            targets_map.extend(targets.signed.signed.targets_map());
+        }
+        targets_map
+    }
+
+    fn consistent_snapshot(&self) -> bool {
+        self.consistent_snapshot
+    }
+}
+
+/// `PathExists` allows the user of our copy/link functions to specify what happens when the target
+/// is being written to a shared targets directory and the file already exists from another repo.
+#[derive(Debug, Deserialize, Clone, Copy)]
+#[serde(rename_all = "kebab-case")]
+pub enum PathExists {
```

Comment:
Could we move all of this back to where it was? It pre-empts all of this target moving/copying code very nicely.